src/crawler/getPolkadotData.ts

```diff
@@ -38,7 +38,7 @@ export async function runCrawlers(provider, types, storage) {
                     payload.call_hash = _datumPair.extrinsic.method.method == 'asMulti' ? blake2AsHex(_datumPair.extrinsic.method.args[3].toHuman()?.toString()!) : _datumPair.extrinsic.method.args[3].toHuman()?.toString()!; // only as_multi will have actual call_data other calls will only have the call_hash
                     // for final approval we send full call data instead of the hash
 
-                    payload.call_data = _datumPair.extrinsic.method.method == 'asMulti' ? _datumPair.extrinsic.method.args[3].toHuman()?.toString()! : (await storage.query({ "item.call_hash": _datumPair.extrinsic.method.args[3].toHuman()?.toString()! }) as Array<any>)[0]?.item.call_data; // We don't get call_data until the final approval call (as_multi), so I think the call_data for other calls would be empty. but in case we had a call_data in DB before from some other final call of as_multi then we'd have its data and we'd add it here.
+                    payload.call_data = _datumPair.extrinsic.method.method == 'asMulti' ? _datumPair.extrinsic.method.args[3].toHuman()?.toString()! : (await storage.find({ "call_hash": _datumPair.extrinsic.method.args[3].toHuman()?.toString()! }) as Array<any>)[0]?.call_data; // We don't get call_data until the final approval call (as_multi), so I think the call_data for other calls would be empty. but in case we had a call_data in DB before from some other final call of as_multi then we'd have its data and we'd add it here.
 
                     payload.status = event.method == 'NewMultisig' && (_datumPair.extrinsic.method.method == 'approveAsMulti' || _datumPair.extrinsic.method.method == 'asMulti') ? 'created' : event.method == 'MultisigApproval' && _datumPair.extrinsic.method.method == 'approveAsMulti' ? 'approving' : event.method == 'MultisigExecuted' && _datumPair.extrinsic.method.method == 'asMulti' ? 'executed' : event.method == 'MultisigCancelled' && _datumPair.extrinsic.method.method == 'cancelAsMulti' ? 'cancelled' : '';
```

Comment:
I don't think you need to check the extrinsic name.
And a line should not be too long, it's hard to read. maybe use if or switch here like this:
```js
switch(event.method) {
  case 'NewMultisig': 
    payload.status = 'created';
    break;
  case 'MultisigApproval':
    payload.status = 'approving';
    break;
  case 'MultisigExecuted':
    payload.status = 'executed';
    break;
  case 'MultisigCancelled':
    payload.status = 'cancelled';
    break;
  default:
    payload.status = ''
}
```

Reply:
I agree, just the event name will do too.
ahh, just realized line wrapping was on in my editor, didn't realize the line length reached 154. :grin: